### DISCUSSION TITLE
Add support for bookmarks in the editor

### DISCUSSION THREAD

**Discussion by 64ArthurAraujo** [ppy/osu#27775] [Ideas]
As i saw in recent discussion and felt the need myself, i would like to implement bookmarks in the editor. As i did not find much discussion about this i opened this to ask: How should this be implemented?

 - Bookmark controls
 Design-wise how should they look like to match lazer's look and feel?
![image](https://github.com/ppy/osu/assets/90941580/38ae9af1-d15e-4fe8-b802-d2e92c9b33f3)

- Displaying them in the timeline
  As preview points are displayed below and and "red lines" are displayed above how should bookmarks show? (maybe cross the entire thing?)
![image](https://github.com/ppy/osu/assets/90941580/7179dd36-98c4-431a-b0fc-851ea5fa55e8)

- Labels 
  As suggested here #12978 it would be cool to do this as well, if so how they should look like?

**Comment by Walavouchey** [comment]
@ppy/team-client may need direction

**Comment by bdach** [comment]
Step zero for doing anything with bookmarks is fixing https://github.com/ppy/osu/issues/20883 so that the editor can actually read them. I don't see any point discussing anything else before that is fixed.

Given that that was tried and got stuck (https://github.com/ppy/osu/pull/23418), I'd say this requires one of us (read: core team) to actually sit down and figure out a direction to fix that issue.

**Reply by 64ArthurAraujo** [reply]
I took a look at #23418 and it seems that along side with reading the bookmarks they also implemented them visually on the timeline, which i find weird because by reading only the title and the description you would think this is only a "back-end" change. I would rather split the bookmark implementation into 3 PRs to properly separate concerns and hopefully make reviewing easier:

 1. Address the reading and storing of bookmarks
 2. Show them visually on the timeline
 3. Add the ability to insert, remove, seek between bookmarks and wipe them

Given that this PR is old and have some issues would we need to start again from scratch?

**Reply by bdach** [reply]
Splitting the PRs is fine but as I said this is going to require @ppy/team-client  to actually decide how to solve point 1. It's going to affect much more than just bookmarks so it's not going to be a plain old feature PR.
